packages/flutter_tools/test/commands.shard/hermetic/ios_analyze_test.dart

```diff
@@ -0,0 +1,163 @@
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:args/command_runner.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/analyze.dart';
+import 'package:flutter_tools/src/commands/ios_analyze.dart';
+import 'package:flutter_tools/src/ios/xcodeproj.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/project_validator.dart';
+import 'package:test/fake.dart';
+import 'package:test/test.dart';
+
+import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
+import '../../src/test_flutter_command_runner.dart';
+
+void main() {
+  group('ios analyze command', () {
+    late BufferLogger logger;
+    late FileSystem fileSystem;
+    late Platform platform;
+    late FakeProcessManager processManager;
+    late Terminal terminal;
+    late AnalyzeCommand command;
+    late CommandRunner<void> runner;
+
+    setUpAll(() {
+      Cache.disableLocking();
+    });
+
+    setUp(() {
+      logger = BufferLogger.test();
+      fileSystem = MemoryFileSystem.test();
+      platform = FakePlatform();
+      processManager = FakeProcessManager.empty();
+      terminal = Terminal.test();
+      command = AnalyzeCommand(
+        artifacts: Artifacts.test(),
+        fileSystem: fileSystem,
+        logger: logger,
+        platform: platform,
+        processManager: processManager,
+        terminal: terminal,
+        allProjectValidators: <ProjectValidator>[],
+        suppressAnalytics: true,
+      );
+      runner = createTestCommandRunner(command);
+
+      // Setup repo roots
+      const String homePath = '/home/user/flutter';
+      Cache.flutterRoot = homePath;
+      for (final String dir in <String>['dev', 'examples', 'packages']) {
+        fileSystem.directory(homePath).childDirectory(dir).createSync(recursive: true);
+      }
+    });
+
+    test('can output json file', () async {
+      final MockIosProject ios = MockIosProject();
+      final MockFlutterProject project = MockFlutterProject(ios);
+      const String expectedConfig = 'someConfig';
+      const String expectedScheme = 'someScheme';
+      const String expectedTarget = 'someConfig';
+      const String expectedOutputFile = '/someFile';
+      ios.outputFileLocation = expectedOutputFile;
+      await IOSAnalyze(
+        project: project,
+        option: IOSAnalyzeOption.outputUniversalLinkSettings,
+        configuration: expectedConfig,
+        scheme: expectedScheme,
+        target: expectedTarget,
+        logger: logger,
+      ).analyze();
+      expect(logger.statusText, contains(expectedOutputFile));
```

Comment:
i think we should test that we're writing the write data to this file.

Reply:
The writing data part is mocked in this test, and the actual functionality is tested  in the IosProject.outputUniversalLinkSettings unit test
https://github.com/flutter/flutter/blob/cf1968aa666e694aa60670a32b7cee556edbe1e3/packages/flutter_tools/test/general.shard/project_test.dart#L727

Reply:
This test is just making sure the analyze end up calling IosProject.outputUniversalLinkSettings with the right parameters

Reply:
sorry for the delay, I will finish reviewing before EOD today (Thursday)